services/transaction-statistics/shared/utils/requestAll.js

```diff
@@ -0,0 +1,51 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2022 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const requestAll = async (fn, method, params, limit) => {
+	const defaultMaxAmount = limit || 1000;
+	const oneRequestLimit = params.limit || 100;
+	const firstRequest = await fn(method,
+		{
+			...params,
+			...{
+				limit: oneRequestLimit,
+				offset: 0,
+			},
+		});
+	const { data } = firstRequest;
+	if (!data.error) {
```

Comment:
Please cross-check if we send out an `error` in response from the connector when we hit the request limits. Else, we'll need to adjust the same.

Reply:
Yes, we always send `error` in response in case of request failure or no data found.